jest.config.types.js

```diff
@@ -7,29 +7,15 @@
 
 'use strict';
 
-const assert = require('assert');
-const baseConfig = require('./jest.config');
-
-const {
-  modulePathIgnorePatterns,
-  testPathIgnorePatterns,
-  watchPathIgnorePatterns,
-} = baseConfig;
-
-assert.strictEqual(
-  testPathIgnorePatterns[0],
-  '/test-types/',
-  'First entry must be types',
-);
+const {modulePathIgnorePatterns} = require('./jest.config');
 
 module.exports = {
   displayName: {
     color: 'blue',
     name: 'types',
   },
   modulePathIgnorePatterns,
+  roots: ['<rootDir>/packages'],
   runner: 'jest-runner-tsd',
-  testMatch: ['<rootDir>/test-types/*.test.ts'],
-  testPathIgnorePatterns: testPathIgnorePatterns.slice(1),
-  watchPathIgnorePatterns,
+  testMatch: ['**/__typechecks__/**/*.ts'],
```

Comment:
I like `typechecks`, but it could be named somehow different too. It is just some distinctive name to simplify the setup.